# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1257877557 18000
#      Tue Nov 10 13:25:57 2009 -0500
# Node ID 7161e9b41cf625a71031dd98e050bd2850be073f
# Parent  384616240a8fef58db84f33f52252abe9a344e7e
__go_draw_axes__.m: No 'with linestyle' when linestyle == 'none'.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-11-10  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: No 'with linestyle' when linestyle == 'none'.
+
 2009-11-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/arrayfun.m: Fix nargout=0 case.
 
 2009-11-08  Petr Mikulik <mikulik@physics.muni.cz>
 
 	* plot/gnuplot_drawnow.m: Support gnuplot's dumb terminal.
 
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -520,18 +520,22 @@ function __go_draw_axes__ (h, plot_strea
 	      usingclause{data_idx} = sprintf ("record=%d using ($1):($2) axes %s%s",
 					      rows(xdat), xaxisloc_using, yaxisloc_using);
 	    endif
 	  endif
 
 	  style = do_linestyle_command (obj, obj.color, data_idx, mono, 
 					plot_stream, errbars);
 
-	  withclause{data_idx} = sprintf ("with %s linestyle %d",
-					  style{1}, data_idx);
+	  if (isempty (style{1}))
+	    withclause{data_idx} = '';
+	  else
+	    withclause{data_idx} = sprintf ("with %s linestyle %d",
+					    style{1}, data_idx);
+	  endif
 
 	  if (length (style) > 1)
 	    data_idx++;
 	    is_image_data(data_idx) = is_image_data(data_idx - 1); 
 	    parametric(data_idx) = parametric(data_idx - 1);
 	    have_cdata(data_idx) = have_cdata(data_idx - 1);
 	    have_3d_patch(data_idx) = have_3d_patch(data_idx - 1);
 	    titlespec{data_idx} = "title \"\"";
